pkg/promtail/promtail_test.go

```diff
@@ -628,3 +633,25 @@ func randName() string {
 	}
 	return string(b)
 }
+
+func Test_DryRun(t *testing.T) {
+
+	f, err := ioutil.TempFile("/tmp", "Test_DryRun")
```

Comment:
I really prefer to use in-memory versions when files aren't necessary, but it's not a deal breaker. After a second look it seems we'd have to do some refactoring for my suggestion to work.